### PR TITLE
fix: allow feature_tag to be optional

### PR DESCRIPTION
**Pull request by duyhungtnn**
# Changes 
- [x] allow feature_tag to be optional
- [x] update e2e test

### CODE REVIEW

File: example/integration_test/bucketeer_flutter_sdk_test.dart [hunk 1]
```diff
@@ -351,4 +355,27 @@ void main() async {
           BKTClient.instance.removeEvaluationUpdateListener(listenToken);
         });
   });
+
+  test('BKTClient.initialize should allow feature_tag to be optional', () async {
```
Comment (cre8ivejp) [inline example/integration_test/bucketeer_flutter_sdk_test.dart]: Please add a unit test and check if the tag is empty after initializing the SDK.

E.g.

```kt
expect(config.featureTag, '');
```

File: example/integration_test/bucketeer_flutter_sdk_test.dart [hunk 2]
```diff
@@ -351,4 +355,27 @@ void main() async {
...
+        .pollingInterval(Constants.examplePollingInterval)
+        .backgroundPollingInterval(Constants.exampleBackgroundPollingInterval)
+        .appVersion(appVersion)
+        .build();
+    final user = BKTUserBuilder().id(userId).customAttributes({}).build();
+
+    await BKTClient.initialize(
+      config: config,
+      user: user,
+    ).onError((error, stackTrace) => fail("initialize() should success"));
```
Comment (cre8ivejp) [inline example/integration_test/bucketeer_flutter_sdk_test.dart]: We should also ensure that are getting the evaluations from the server correctly.
Use some get variation interface to check it.

File: test/bucketeer_config_builder_test.dart [hunk 3]
```diff
@@ -19,6 +19,7 @@ void main() {
         .appVersion("1.0.0");
     expect(() => builderMissingAPIEndpoint.build(), throwsA(isA<ArgumentError>()));
 
+    // FeatureTag could be optional
     final builderMissingFeatureTag = BKTConfigBuilder()
```
Comment (cre8ivejp) [inline test/bucketeer_config_builder_test.dart]: nit: builderWithEmptyFeatureTag.

File: example/integration_test/bucketeer_flutter_sdk_test.dart [hunk 4]
```diff
@@ -322,33 +329,72 @@ void main() async {
...
-          BKTClient.instance.removeEvaluationUpdateListener(listenToken);
-        });
+      // Check remove the listener. If the `removeEvaluationUpdateListener` fail, the test will fail.
+      // The `completer` instance may get more call more times.
+      // Because it already complete, it will throw an exception cause the test fail.
+      BKTClient.instance.removeEvaluationUpdateListener(listenToken);
+    });
+  });
+
+  test('BKTClient should allow feature_tag to be optional',
```
Comment (cre8ivejp) [inline example/integration_test/bucketeer_flutter_sdk_test.dart:362]: Did the e2e test pass?

Comment (duyhungtnn) [reply example/integration_test/bucketeer_flutter_sdk_test.dart:362]: the Android e2e passed. I am looking on the iOS

Comment (duyhungtnn) [reply example/integration_test/bucketeer_flutter_sdk_test.dart:362]: #### iOS e2e test passed

Screenshot 2023-10-17 at 15 03 34

#### Android e2e test passed


Screenshot 2023-10-17 at 15 10 41

**Review by cre8ivejp — APPROVED** [review]
Thank you!
